auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -97,6 +99,10 @@ public Property(String humanName, ExecutableElement element) {
       annotations = buildAnnotations(element);
 
       typeAdapter = getAnnotationValue(element, GsonTypeAdapter.class);
+      TypeMirror returnType = element.getReturnType();
+      instanceOfOptional = !TypeKind.TYPEVAR.equals(returnType.getKind())
```

Comment:
This is a bit limited since it only supports guava and java 8's. Could we make it more generic? (Unless AutoValue only supports these two, then it's fine. My memory's fuzzy)